From 0882db22b0ab0fbf192529ef76969bb691360529 Mon Sep 17 00:00:00 2001
From: Alaa Hleihel <ahleihel@redhat.com>
Date: Tue, 12 May 2020 10:55:24 -0400
Subject: [PATCH 207/312] [netdrv] net/mlx5e: en_accel, Add missing
 net/geneve.h include

Message-id: <20200512105530.4207-119-ahleihel@redhat.com>
Patchwork-id: 306991
Patchwork-instance: patchwork
O-Subject: [RHEL8.3 BZ 1789382 118/124] net/mlx5e: en_accel, Add missing net/geneve.h include
Bugzilla: 1789382
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Kamal Heib <kheib@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Bugzilla: http://bugzilla.redhat.com/1789382
Upstream: net-next/master

commit 9425c595bd513948537ef355c07a65595dd2c771
Author: Raed Salem <raeds@mellanox.com>
Date:   Mon Jan 20 15:03:00 2020 +0200

    net/mlx5e: en_accel, Add missing net/geneve.h include

    The cited commit relies on include <net/geneve.h> being included
    implicitly prior to include "en_accel/en_accel.h".
    This mandates that all files that needs to include en_accel.h
    to redantantly include net/geneve.h.

    Include net/geneve.h explicitly at "en_accel/en_accel.h" to avoid
    undesired constrain as above.

    Fixes: e3cfc7e6b7bd ("net/mlx5e: TX, Add geneve tunnel stateless offload support")
    Signed-off-by: Raed Salem <raeds@mellanox.com>
    Reviewed-by: Tariq Toukan <tariqt@mellanox.com>
    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>

Signed-off-by: Alaa Hleihel <ahleihel@redhat.com>
Signed-off-by: Frantisek Hrbata <fhrbata@redhat.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/en_accel/en_accel.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/en_accel/en_accel.h b/drivers/net/ethernet/mellanox/mlx5/core/en_accel/en_accel.h
index 3022463f2284..a6f65d4b2f36 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/en_accel/en_accel.h
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_accel/en_accel.h
@@ -42,6 +42,8 @@
 #include "en/txrx.h"
 
 #if IS_ENABLED(CONFIG_GENEVE)
+#include <net/geneve.h>
+
 static inline bool mlx5_geneve_tx_allowed(struct mlx5_core_dev *mdev)
 {
 	return mlx5_tx_swp_supported(mdev);
-- 
2.13.6

